src/utils/__tests__/async.test.ts

```diff
@@ -4,6 +4,41 @@ import { logger } from '../../logger';
 
 jest.mock('../../logger');
 
+import { retrySpawnProcess } from '../async';
+import { spawnProcess } from '../system';
+
+jest.mock('../system', () => {
+  const original = jest.requireActual('../system');
+  return {
+    ...original,
+    spawnProcess: jest.fn(original.spawnProcess),
+  };
+});
+
+describe('retrySpawnProcess', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('resolves before max retries', async () => {
+    const res = await retrySpawnProcess('ls', [], { maxRetries: 1 });
+    expect(res).toBeInstanceOf(Buffer);
+  });
+
+  test('hits max retries and exits', async () => {
```

Comment:
Where are my tests that are actually testing the retry delay, exponential backoff etc? You can get some inspiration from `withRetry` tests.

Reply:
I think this comment still holds but I'll give you a pass hoping you'll add these tests in the near future 🙂